From e4d8c6da0fc9e8ba98430fb176c62c7a4a542079 Mon Sep 17 00:00:00 2001
From: "Angelo G. Del Regno" <kholk11@gmail.com>
Date: Fri, 12 Jul 2019 12:42:44 +0200
Subject: [PATCH 127/175] clk: qcom: aop-qmp: Add support for v1 AOP clocks

Due to HW limitations on v1, the qdss_ao clock was not supported
by the clock driver on AOP.

This is needed for SDM845.
---
 drivers/clk/qcom/clk-aop-qmp.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/clk/qcom/clk-aop-qmp.c b/drivers/clk/qcom/clk-aop-qmp.c
index 0f7b77d01228..3956cfb41726 100644
--- a/drivers/clk/qcom/clk-aop-qmp.c
+++ b/drivers/clk/qcom/clk-aop-qmp.c
@@ -256,6 +256,15 @@ static struct notifier_block aop_qmp_clk_panic_notifier = {
 	.priority = 1,
 };
 
+/*
+ * Due to HW limitations on v1, the qdss_ao clock was not supported by the clock
+ * driver on AOP.
+ */
+static void aop_qmp_fixup_v1(void)
+{
+	aop_qmp_clk_hws[QDSS_AO_CLK] = NULL;
+}
+
 static int qmp_update_client(struct clk_hw *hw, struct device *dev,
 		struct mbox_chan **mbox)
 {
@@ -314,6 +323,9 @@ static int aop_qmp_clk_probe(struct platform_device *pdev)
 
 	clk_data->clk_num = num_clks;
 
+	if (of_device_is_compatible(pdev->dev.of_node, "qcom,aop-qmp-clk-v1"))
+		aop_qmp_fixup_v1();
+
 	for (i = 1; i < num_clks; i++) {
 		if (!aop_qmp_clk_hws[i])
 			continue;
@@ -366,6 +378,7 @@ static int aop_qmp_clk_probe(struct platform_device *pdev)
 
 static const struct of_device_id aop_qmp_clk_of_match[] = {
 	{ .compatible = "qcom,aop-qmp-clk", },
+	{ .compatible = "qcom,aop-qmp-clk-v1" },
 	{}
 };
 
-- 
2.48.1

